semigroupoids.cabal

```diff
@@ -204,6 +209,9 @@ library
   if impl(ghc >= 7.10)
     ghc-options: -fno-warn-trustworthy-safe
```

Comment:
I just remembered that I disable this Safe Haskell–related warning some time ago, since I was unable to fix the warning on all supported versions of GHC without horrible amounts of CPP. Does this PR change thing in this regard?

Reply:
I changed it for `contravariant`, I could change it here too, i.e.
- remove `-fno-warn-trustworthy-safe`
- annotate modules to be `Trustworthy/Safe`.

But OTOH, it doesn't matter much until someone actually audits trustworthyness. (`Safe` modules don't need to be checked, at least in theory).

Reply:
I don't have a strong opinion on the matter—do whatever is easiest. I just wanted to make sure you were aware of this.

Reply:
I'll rather keep changes as small as reasonable (`contravariant` has few modules, `semigroupoids` doesn't).